src/App.tsx

```diff
@@ -2,72 +1,118 @@
+import React, { useEffect, useState, useRef } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [inputValue, setInputValue] = useState('');
+  const [filteredPeople, setFilteredPeople] = useState(peopleFromServer);
+  const [selectedPerson, setSelectedPerson] = useState<null | {
+    name: string;
+    born: number;
+    died: number;
+  }>(null);
+  const [isDropdownVisible, setIsDropdownVisible] = useState(false);
+  const [debounceTimer, setDebounceTimer] = useState<NodeJS.Timeout | null>(
+    null,
+  );
+
+  const dropdownRef = useRef<HTMLDivElement | null>(null);
+
+
+  const debounceDelay = 300;
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const value = e.target.value;
+
+    setInputValue(value);
+    setSelectedPerson(null);
+    setIsDropdownVisible(true);
+
+    if (debounceTimer) {
+      clearTimeout(debounceTimer);
+    }
+
+    const newTimer = setTimeout(() => {
+      if (value === '') {
+        setFilteredPeople(peopleFromServer);
+      } else {
+        setFilteredPeople(
+          peopleFromServer.filter(person =>
+            person.name.toLowerCase().includes(value.toLowerCase()),
+          ),
+        );
+      }
+    }, debounceDelay);
+
+    setDebounceTimer(newTimer);
```

Comment:
Ensure that filtering does not occur if the text hasn't changed. You might want to compare the current input value with the previous one before setting the debounce timer.